src/main/java/seedu/address/logic/commands/modulelistcommands/EditModuleCommand.java

```diff
@@ -95,10 +96,12 @@ private static Module createEditedModule(Module moduleToEdit, EditModuleDescript
         // Address updatedAddress = editModuleDescriptor.getAddress().orElse(ModuleToEdit.getAddress());
         Set<Tag> updatedTags = editModuleDescriptor.getTags().orElse(moduleToEdit.getTags());
         // return new Module(updatedName, updatedEmail, updatedAddress, updatedTags);
-
         ModuleName moduleName = editModuleDescriptor.getModuleName().orElse(moduleToEdit.getName());
         ZoomLink zoomLink = editModuleDescriptor.getZoomLink().orElse(moduleToEdit.getLink());
-        GradeTracker gradeTracker = editModuleDescriptor.getGradeTracker().orElse((moduleToEdit.getGradeTracker()));
+        GradeTracker gradeTracker = gradeTracker = moduleToEdit.getGradeTracker();
+        if (editModuleDescriptor.gradePoint != null) {
+            gradeTracker.setGradePoint(Optional.of(editModuleDescriptor.gradePoint));
+        }
```

Comment:
Perhaps a getter method for gradePoint can be added? To preserve consistency since all the other fields are accessed with getters.